src/moodleapi/Moodleapi.ts

```diff
@@ -0,0 +1,39 @@
+interface Response {
+  uuid: string
+  jwt: string
+}
+export class Moodleapi {
+  endpoint = 'local_raise_get_user'
+
+  async getUser(): Promise<Response> {
+    const userRequest = {
+      index: 0,
+      methodname: this.endpoint,
+      args: {
+      }
+    }
+    const url = this.getMoodlePluginMethodUrl(this.endpoint)
+    if (url === '') {
+      console.log('Invalid environment variables')
+    }
+    const response = await fetch(url, { method: 'POST', body: JSON.stringify([userRequest]) })
+    const responseJSON = await response.json()
+
+    return {
+      uuid: responseJSON[0].data.uuid,
+      jwt: responseJSON[0].data.jwt
+    }
+  }
+
+  getMoodlePluginMethodUrl(endpointMethod: string): string {
+    if (window.M === undefined) {
+      return ''
+    }
```

Comment:
Ah, I think this type of logic is why I was thinking earlier we should pass the root and sesskey to the constructor. That way any logic around whether we can even talk to Moodle happens outside by the caller and none of this needs to be here. Instead the below can be:

```javascript
    let userUrl: string
    userUrl = this.wwwroot
    userUrl += '/lib/ajax/service.php?sesskey='
    userUrl += this.sesskey
    userUrl += '&info=' + endpointMethod
    return userUrl
```

Perhaps the constructor then looks like:

```javascript
  private readonly wwwroot: string
  private readonly sesskey: string

  constructor(wwwroot: string, sesskey: string) {
    this.wwwroot = wwwroot
    this.sesskey = sesskey
  }
```